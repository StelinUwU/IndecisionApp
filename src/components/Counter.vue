<template>
  <h2>{{ title }}</h2>
  <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-testid="counter">{{ counter }}</p>

  <div>
    <button @click="increment">+1</button>
    <button @click="decrement">-1</button>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: 'Counter',
    },
    start: {
      type: Number,
      default: 0,
      validator: (value) => value >= 0,
    },
  },

  data() {
    return {
      counter: this.start,
    };
  },

  methods: {
    increment() {
      this.counter++;
    },
    decrement() {
      this.counter--;
    },
  },

  computed: {
    squareCounter() {
      return this.counter * this.counter;
    },
  },
};
</script>

<style scoped>
button {
  background-color: #64bb87;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  margin: 0 5px;
  transition: 0.3s ease-in-out;
}
</style>
