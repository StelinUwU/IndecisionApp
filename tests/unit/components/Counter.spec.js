import Counter from '@/components/Counter';
import { shallowMount } from '@vue/test-utils';
describe('Counter component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Counter);
  });

  test('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should have the default value', () => {
    expect(wrapper.find('h2').exists()).toBeTruthy();

    const h2Value = wrapper.find('h2').text();

    expect(h2Value).toBe('Counter');
  });

  test('Second paragraph default value should be 0', () => {
    const pValue = wrapper.find('[data-testid=counter]').text();

    expect(pValue).toBe('0');
  });

  test('should increase by 1 the counter value', async () => {
    let pValue;

    const [increaseBtn, decreaseBtn] = wrapper.findAll('button');

    await increaseBtn.trigger('click');
    await increaseBtn.trigger('click');

    await decreaseBtn.trigger('click');

    pValue = wrapper.find('[data-testid=counter]').text();

    expect(pValue).toBe('1');
  });

  test('should set the default value', () => {
    const { start } = wrapper.props();

    const value = wrapper.find('[data-testid=counter]').text();

    expect(value).toBe(`${start}`);
  });

  test('should show the title prop', () => {
    const title = 'Counter title';

    const wrapper = shallowMount(Counter, {
      props: {
        title,
      },
    });

    const h2Value = wrapper.find('h2').text();

    expect(h2Value).toBe(title);
  });
});
