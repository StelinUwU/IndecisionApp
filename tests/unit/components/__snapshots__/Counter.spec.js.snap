// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match snapshot 1`] = `
<h2>Counter</h2>
<p>0 <sup>2</sup> = 0</p>
<p data-testid="counter">0</p>
<div><button>+1</button><button>-1</button></div>
`;
